### PR TITLE
Add initial Java documentation

### PR DESCRIPTION
**Pull request by flands**
Addresses #283 This will supersede documentation currently captured in README. Separate
PRs will address this.

TODOs in future PRs:
- Exporters
- Sampling

### CODE REVIEW

File: content/en/docs/java/manual_instrumentation.md [hunk 1]
```diff
@@ -0,0 +1,372 @@
+---
+title: "Manual Instrumentation"
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:2]:
```suggestion
Title: "Manual Instrumentation"
```

File: content/en/docs/java/manual_instrumentation.md [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+weight: 3
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:3]:
```suggestion
Weight: 3
```

File: content/en/docs/java/manual_instrumentation.md [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+In order to instrument, a `Tracer` must be acquired, which is responsible for
+creating spans and interacting with the [`Context`](#context-propagation). A
+`Tracer` is acquired by using the [OpenTelemetry Tracing
+API](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md)
+specifying the name and version of the [library
+instrumenting](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/glossary.md#instrumentation-library)
+the [instrumented
+library](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/glossary.md#instrumented-library)
+or application to be monitored.
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:31]:
```suggestion
In order to instrument, you must acquire a `Tracer`. A `Tracer` is responsible for
creating spans and interacting with the [`Context`](#context-propagation). To acquire a
`Tracer` use the [OpenTelemetry Tracing
API](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md)
and specify the name and version of the [library
instrumenting](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/glossary.md#instrumentation-library)
the [instrumented
library](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/glossary.md#instrumented-library)
or application to be monitored.
```

File: content/en/docs/java/manual_instrumentation.md [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+} finally {
+    span.end(); // closing the scope does not end the span, this has to be done manually
+}
+```
+
+#### Nested
+
+Most of the time, we want to correlate spans for nested operations.
+OpenTelemetry supports tracing within processes and across remote processes.
+For more details how to share context between remote processes, see [Context
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:61]:
```suggestion
For more information about how to share context between remote processes, see [Context
```

File: content/en/docs/java/manual_instrumentation.md [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+void b(Span parentSpan) {
+  Span childSpan = tracer.spanBuilder("b")
+        .setParent(parentSpan)
+        .startSpan();
+  // do stuff
+  childSpan.end();
+}
+```
+
+The OpenTelemetry API offers also an automated way to propagate the parentSpan:
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:83]:
```suggestion
The OpenTelemetry API also offers an automated way to propagate the parentSpan:
```

File: content/en/docs/java/manual_instrumentation.md [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+```
+
+### Enrich Spans
+
+#### Attributes
+
+In OpenTelemetry spans can be created freely and it's up to the implementor to
+annotate them with attributes specific to the represented operation. Attributes
+provide additional context on a span about the specific operation it tracks,
+such as results or operation properties.
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:121]:
```suggestion
In OpenTelemetry, you can create spans freely. It's up to the implementor to
annotate spans with attributes specific to the represented operation. Attributes
provide additional context on a span about the specific operation it tracks,
such as results or operation properties.
```

File: content/en/docs/java/manual_instrumentation.md [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+
+```java
+Span span = tracer.spanBuilder("/resource/path").setSpanKind(Span.Kind.CLIENT).startSpan();
+span.setAttribute("http.method", "GET");
+span.setAttribute("http.url", url.toString());
+```
+
+Some of these operations represent calls that use well-known protocols like
+HTTP or database calls. For these, OpenTelemetry requires specific attributes
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:130]:
```suggestion
HTTP or database calls. For these operations, OpenTelemetry requires specific attributes
```

File: content/en/docs/java/manual_instrumentation.md [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+
+span.addEvent("End Computation", eventAttributes);
+```
+
+#### Links
+
+A Span may be linked to zero or more other Spans that are causally related.
+Links can be used to represent batched operations where a Span was initiated by
+multiple initiating Spans, each representing a single incoming item being
+processed in the batch.
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:159]:
```suggestion
You can link a span to zero or more other spans that are causally related.
Use links to represent batched operations where a span was initiated by
multiple initiating spans, each representing a single incoming item being
processed in the batch.
```

File: content/en/docs/java/manual_instrumentation.md [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+```java
+Span child = tracer.spanBuilder("childWithLink")
+        .addLink(parentSpan1.getContext())
+        .addLink(parentSpan2.getContext())
+        .addLink(parentSpan3.getContext())
+        .addLink(remoteContext)
+    .startSpan();
+```
+
+For more details how to read context from remote processes, see [Context
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:170]:
```suggestion
For more information about how to read context from remote processes, see [Context
```

File: content/en/docs/java/manual_instrumentation.md [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+Propagation](#context-propagation).
+
+### Context Propagation
+
+OpenTelemetry provides a text-based approach to propagate context to remote
+services. By default, the [W3C Trace Context](https://www.w3.org/TR/trace-context/)
+format is used.
+
+The following presents an example of an outgoing HTTP request using
+`HttpURLConnection`.
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:180]:
```suggestion
The following example shows an outgoing HTTP request using
`HttpURLConnection`:
```

File: content/en/docs/java/manual_instrumentation.md [hunk 11]
```diff
@@ -0,0 +1,372 @@
...
+} finally {
+  outGoing.end();
+}
+...
+```
+
+Similarly, the text-based approach can be used to read the W3C Trace Context
+from incoming requests. The following presents an example of processing an
+incoming HTTP request using
+[HttpExchange](https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpExchange.html).
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:213]:
```suggestion
Similarly, you can use the text-based approach to read the W3C Trace Context
from incoming requests. The following example demonstrates processing an
incoming HTTP request using
[HttpExchange](https://docs.oracle.com/javase/8/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpExchange.html):
```

File: content/en/docs/java/manual_instrumentation.md [hunk 12]
```diff
@@ -0,0 +1,372 @@
...
+[Zipkin
+B3](https://github.com/open-telemetry/opentelemetry-java/tree/master/extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation).
+
+### Processors
+
+The following processors are available today:
+
+#### SimpleSpanProcessor
+
+This span processor exports span immediately after they are ended.
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:261]:
```suggestion
This span processor exports spans immediately after they end.
```

File: content/en/docs/java/manual_instrumentation.md [hunk 13]
```diff
@@ -0,0 +1,372 @@
...
+
+Example:
+
+```java
+BatchSpanProcessor batchSpansProcessor =
+    BatchSpanProcessor.builder(exporter).build();
+tracerProvider.addSpanProcessor(batchSpansProcessor);
+```
+
+A varitey of configuration parameters can also be specified:
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:282]:
```suggestion
You can also specify a variety of configuration parameters:
```

File: content/en/docs/java/manual_instrumentation.md [hunk 14]
```diff
@@ -0,0 +1,372 @@
...
+        .setExporterTimeoutMillis(
+            30_000) // max amount of time an export can run before getting interrupted
+        .setScheduleDelayMillis(5000) // set time between two different exports
+        .build();
+tracerProvider.addSpanProcessor(batchSpansProcessor);
+```
+
+#### MultiSpanProcessor
+
+A MultiSpanProcessor accepts a list of Span Processors.
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:299]:
```suggestion
A MultiSpanProcessor accepts a list of span processors.
```

File: content/en/docs/java/manual_instrumentation.md [hunk 15]
```diff
@@ -0,0 +1,372 @@
...
+*TODO*
+
+## Metrics
+
+OpenTelemetry provides support for metrics, a time series of numbers that might
+express things such as CPU utilization, request count for an HTTP server or a
+business metric such as transactions.
+
+All metrics can be annotated with labels: additional qualifiers that help
+describe what subdivision of the measurements the metric represents.
```
Comment (snyder114) [inline content/en/docs/java/manual_instrumentation.md:324]:
```suggestion
OpenTelemetry provides support for metrics, a time series of numbers that might
express things such as CPU utilization, request count for an HTTP server, or a
business metric such as transactions.

All metrics can be annotated with labels. Labels are additional qualifiers that help
describe what subdivision of the measurements the metric represents.
```

File: content/en/docs/java/getting_started.md [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+extract the context and create a child span with an attached span event.
+All generated data is logged locally.
+
+To get started:
+
+- Clone the respository
+- Change into the example directory
+- Compile the code
+- Start the server
+- Start the client
```
Comment (snyder114) [inline content/en/docs/java/getting_started.md:107]:
```suggestion
To get started, perform the following steps::

1. Clone the repository
2. Change into the example directory
3. Compile the code
4. Start the server
5. Start the client
```

**Review by snyder114 — CHANGES_REQUESTED** [review]

**Comment by flands** [comment]
@snyder114 thanks for the feedback! I noticed you recently merged an updated README to the Java repo. FYI, it will be removed once this PR is merged and then https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1581 is merged. Would love to get the changes added here as well.

**Review by austinlparker — APPROVED** [review]
lgtm - could you fix the broken links?

File: content/en/docs/java/getting_started.md [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+
+An exporter jar can be specified via the `otel.exporter.jar` system property:
+
+```java
+java -javaagent:path/to/opentelemetry-javaagent-all.jar \
+     -Dotel.exporter.jar=path/to/external-exporter.jar \
+     -jar myapp.jar
+```
+
+For additional information, see the [Automatic Instrumentation documentation](/docs/java/automatic-instrumentation/#traces).
```
Comment (flands) [inline content/en/docs/java/getting_started.md:44]:
```suggestion
For additional information, see the [Automatic Instrumentation documentation](/docs/java/automatic_instrumentation).
```

File: content/en/docs/java/getting_started.md [hunk 18]
```diff
@@ -0,0 +1,161 @@
...
+      Thread.sleep(100);
+    } finally {
+      // End span
+      exampleSpan.end();
+    }
+  }
+}
+```
+
+For more detailed information, see the [Manual Instrumentation documentation](/docs/java/manual-instrumentation/#traces).
```
Comment (flands) [inline content/en/docs/java/getting_started.md:87]:
```suggestion
For more detailed information, see the [Manual Instrumentation documentation](/docs/java/manual_instrumentation/#traces).
```

File: content/en/docs/java/getting_started.md [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
+
+  for (int i = 0; i < 10; i++) {
+    // Create a metric
+    counter.add(1);
+  }
+
+}
+```
+
+For more detailed information, see the [Manual Instrumentation documentation](/docs/java/manual-instrumentation/#metrics).
```
Comment (flands) [inline content/en/docs/java/getting_started.md:153]:
```suggestion
For more detailed information, see the [Manual Instrumentation documentation](/docs/java/manual_instrumentation/#metrics).
```

File: content/en/docs/java/automatic_instrumentation.md [hunk 20]
```diff
@@ -0,0 +1,389 @@
...
+This is useful in case you have code that is over-instrumented using `@WithSpan`,
+and you want to suppress some of them without modifying the code.
+
+| System property                 | Environment variable            | Purpose                                                                                                                                  |
+|---------------------------------|---------------------------------|------------------------------------------------------------------------------------------------------------------------------------------|
+| trace.annotated.methods.exclude | TRACE_ANNOTATED_METHODS_EXCLUDE | Suppress `@WithSpan` instrumentation for specific methods, format is "my.package.MyClass1[method1,method2];my.package.MyClass2[method3]" |
+
+### Configure the OpenTelemetry getTracer
+
+See the [Manual Instrumentation documentation](/docs/java/manual-instrumentation/#spans)
```
Comment (flands) [inline content/en/docs/java/automatic_instrumentation.md:379]:
```suggestion
See the [Manual Instrumentation documentation](/docs/java/manual_instrumentation/#instantiate-tracer)
```

File: content/en/docs/java/manual_instrumentation.md [hunk 21]
```diff
@@ -0,0 +1,372 @@
...
+
+## Traces
+
+### Instantiate `TracerProvider`
+
+In the [OpenTelemetry Tracing
+API](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md),
+the `TracerProvider` is the entry point and is expected to be the stateful
+object that holds any configuration. The `TracerProvider` also provides access
+to the [`Tracer`](#instantiate-a-tracer).
```
Comment (flands) [inline content/en/docs/java/manual_instrumentation.md:14]:
```suggestion
to the [`Tracer`](#instantiate-tracer).
```

File: content/en/docs/java/manual_instrumentation.md [hunk 22]
```diff
@@ -0,0 +1,372 @@
...
+
+Some of these operations represent calls that use well-known protocols like
+HTTP or database calls. For these operations, OpenTelemetry requires specific attributes
+to be set. The full attribute list is available in the Semantic Conventions in
+the cross-language specification.
+
+#### Events
+
+Spans can be annotated with named events that can carry zero or more [Span
+Attributes](#span-attributes), each of which is itself a name/value map paired
```
Comment (flands) [inline content/en/docs/java/manual_instrumentation.md:137]:
```suggestion
Attributes](#attributes), each of which is itself a name/value map paired
```

**Pull request merged**